pytest 6: pylint shows not-callable error for builtin pytest marks

With the current versions of pylint (2.5.3) and the underlying astroid (2.4.2), running pylint against a simple pytest 6.0.0rc1 test:

```python3
import pytest

@pytest.mark.parametrize('a', [1, 2])
def test_a(a):
    pass
```

results in:

```
************* Module test_pylint
test_pylint.py:3:1: E1102: pytest.mark.parametrize is not callable (not-callable)
```

which isn't the case with pytest 5.4.3. I was able to bisect this to b3fb5a2d47743a09c551555da22da27ce9e73f41 (#7379, "Type annotate pytest.mark.* builtin marks" by @bluetech).

Note that to my knowledge, pylint doesn't know about type annotations and relies on its inference only. Thus, I suppose it either only sees `parametrize = None`, or it sees `_ParametrizeMarkDecorator.__call__` raising `NotImplementedError`. Not sure what, though. Maybe @PCManticore can explain what exactly is going on there?

Not sure how to best make this work in a way that pylint understands it (or, alternatively, fix it in pylint?) though.

--- a/src/_pytest/mark/structures.py
+++ b/src/_pytest/mark/structures.py
@@ -4,6 +4,7 @@ import typing
 import warnings
 from typing import Any
 from typing import Callable
+from typing import cast
 from typing import Iterable
 from typing import List
 from typing import Mapping
@@ -467,12 +468,14 @@ class MarkGenerator:
 
     # See TYPE_CHECKING above.
     if TYPE_CHECKING:
-        skip = None  # type: _SkipMarkDecorator
-        skipif = None  # type: _SkipifMarkDecorator
-        xfail = None  # type: _XfailMarkDecorator
-        parametrize = None  # type: _ParametrizeMarkDecorator
-        usefixtures = None  # type: _UsefixturesMarkDecorator
-        filterwarnings = None  # type: _FilterwarningsMarkDecorator
+        # Using casts instead of type comments intentionally - issue #7473.
+        # TODO(py36): Change to builtin annotation syntax.
+        skip = cast(_SkipMarkDecorator, None)
+        skipif = cast(_SkipifMarkDecorator, None)
+        xfail = cast(_XfailMarkDecorator, None)
+        parametrize = cast(_ParametrizeMarkDecorator, None)
+        usefixtures = cast(_UsefixturesMarkDecorator, None)
+        filterwarnings = cast(_FilterwarningsMarkDecorator, None)
 
     def __getattr__(self, name: str) -> MarkDecorator:
         if name[0] == "_":

